src/java.base/share/classes/java/lang/runtime/ExactConversionsSupport.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.lang.runtime;
+
+/**
+ * A casting conversion is considered to be exact if, after applying the conversion
+ * to the value, no exception was raised or loss of information has occured;
+ * otherwise it is considered inexact.
+ * The methods in this class provide the run-time support for primitive conversions exactness checks.
+ * These methods may be used, for example, by Java compiler implementations to implement checks
+ * for instanceof and pattern matching runtime implementations.
+ * See JLS section 5.5.1 for more information on exact casting conversions.
+ *
+ * @jls 5.5.1 Exact Casting Conversions
+ * @jls 15.20.2 The instanceof Operator
+ *
+ * @since 22
+ */
+public final class ExactConversionsSupport {
+
+    private ExactConversionsSupport() { }
+
+     /** Exactness method from int to byte
```

Comment:
nit: there seems to be an indentation issue in these javadoc, the subsequent lines should be aligned with the first one. What I see in other javadocs is that we don't write in the first line after the `/**` and instead the javadoc starts in the subsequent line.

side: would it make sense to generate this file mechanically? seems like a great candidate for that